lib.py

```diff
@@ -0,0 +1,305 @@
+import time
+from telebot import types
+import yaml
+from pathlib import Path
+import logging
+from itertools import zip_longest
+import random
+logger = logging.getLogger(__name__)
+import threading
+lock = threading.Lock()
+
+
+#TODO
+# ----1) сделать отдельный файл с объявлениями ф-ий (
+# 2) PyTest (или ст питоновский юнит тест import unittest (test.py)
+# 3) ПРОБЛЕМА прога зависит от того, когда я её запустил - нужен график выполнения функции round_create
+# ----4) JSON либо yaml - под хранение БД - пока через txt до прода доведу лучше
+# ----5) сделать write_active_participants()
+# ----6) что если один чел в конце останется?
+# 7) если YAML пустой на моменте вызова ф-ии read(), то всё падает
+# ----8) file threads safety! Mutex?
+# 9) в main() процесс не идёт дальше bot.poling
+# 10) create random_pairing()
+
+
+def read_users():
+    """
+    read users from db/users.yaml
+    """
+    # users = yaml.safe_load(str(Path('db', 'users.yaml')))
+    with lock:
```

Comment:
We need another mutex for preventing a race condition, where one user could prevent another from registering/deleting herself. (comment 2/5)
```suggestion
    with file_lock:
```